tutorials/lava/lib/dl/slayer/pilotnet/pilotnet_dataset.py

```diff
@@ -39,26 +39,51 @@ class PilotNetDataset(Dataset):
     """
     def __init__(
         self, path='data', sequence=16,
-        train=True, visualize=False, transform=None, extract=True
+        train=True, visualize=False, transform=None,
+        extract=True, download=True,
     ):
         self.path = path + '/driving_dataset/'
 
-        dataset_link = 'https://drive.google.com/file/d/'\
-            '0B-KJCaaF7elleG1RbzVPZWV4Tlk/view?usp=sharing'
+        id = '1Ue4XohCOV5YXy57S_5tDfCVqzLr101M7'
+        dataset_link = 'https://docs.google.com/uc?export=download&id={id}'
         download_msg = f'''Please download dataset form \n{dataset_link}')
         and copy driving_dataset.zip to {path}/
-        Note: create the folder if it does not exist.'''.replace(' '*8, '')
+        Note: create the folder if it does not exist.'''.replace(' ' * 8, '')
 
         # check if dataset is available in path. If not download it
         if len(glob.glob(self.path)) == 0:
+            if download is True:
+                os.makedirs(path, exist_ok=True)
+
+                print('Dataset not available locally. Starting download ...')
+                download_cmd = 'wget --load-cookies /tmp/cookies.txt '\
+                    + '"https://docs.google.com/uc?export=download&confirm='\
+                    + '$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate '\
+                    + f"'https://docs.google.com/uc?export=download&id={id}' -O- | "\
+                    + f"sed -rn \'s/.*confirm=([0-9A-Za-z_]+).*/\\1\\n/p\')&id={id}"\
+                    + f'" -O {path}/driving_dataset.zip && rm -rf /tmp/cookies.txt'
```

Comment:
The assumption here is that the system is Linux and that wget, unzip are installed. There are python libraries that could handle the download and extraction allowing cross platform use and dependency management. Assuming Linux may not be a reasonable assumption but perhaps if we documented the platform limitation, this would be fine.

Reply:
I agree. Automatic download only works for Linux for now because of google drive quirks for large files. The current behavior is it tries to run it. If it fails, it raises instructions to fix it manually (which is invoked if the command fails).